File: python/rna-transcription/rna_transcription.py
def to_rna(dna):
    t = {'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U'}

    try:
        return ''.join([t[n] for n in dna])
    except KeyError:
        return ''

File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function () {};


DnaTranscriber.prototype.toRna = function (dna) {
  var dictionary = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U',
  }
  dna = dna.split('')
  rna = dna.map(nucleotide => dictionary[nucleotide])

  if (rna.indexOf(undefined) != -1) {
    throw Error('Invalid input');
  } else {
    rna = rna.join('')
  }

  return rna;
};


module.exports = DnaTranscriber;
